posts/2020-10-23-beta-20.0.0.12.adoc

```diff
@@ -0,0 +1,440 @@
+
+==== mp.jwt.verify.audiences
+
+The `mp.jwt.verify.audiences` property allows you to create list of allowable audience (`aud`) values. At least one of these must be found in the claim. Previously, this had to be configured in the `server.xml` file. Now, you can configure the audiences in the MicroProfile Config property as follows:
+
+[source, java]
+----
+mp.jwt.verify.audiences=conferenceService
+----
+
+==== mp.jwt.verify.publickey.algorithm
+
+The `mp.jwt.verify.publickey.algorithm` property allows you to control the Public Key Signature Algorithm that is supported by the MicroProfile JWT endpoint. The default value is RSA256 if not specified. Previously, this had to be configured in the `server.xml` file. Now, you can configure the public key algorithm used for verification of the JWT in the MicroProfile Config property as follows:
+
+[source, java]
+----
+mp.jwt.verify.publickey.algorithm=ES256
+----
+
+The specification also adds support for the ES256 signature algorithm, while this Open Liberty beta supports using the RS384, RS512, HS384, HS512, ES256, ES384, and the ES512 signature algorithms.
+
+
+To enable the the MicroProfile JWT 1.2 feature in your app, add the following to your `server.xml`:
+
+[source, xml]
+----
+   <featureManager>
+       <feature>mpJwt-1.2</feature>
+   </featureManager>
+----
+
+For more information:
+
+link:https://github.com/eclipse/microprofile-jwt-auth/blob/master/spec/src/main/asciidoc/release-notes.asciidoc#release-notes-for-microprofile-jwt-12[Release Notes for MicroProfile JWT 1.2]
+
+link:https://github.com/eclipse/microprofile-jwt-auth/blob/master/spec/src/main/asciidoc/microprofile-jwt-auth-spec.asciidoc[MicroProfile JWT 1.2 Specification]
+
+[#lra]
+=== Support for MicroProfile LRA
+
+Long Running Actions (LRA) provide a cloud native transaction model based on a compensating Saga system. This allows loosely coupled services to coordinate long running activities, guaranteeing eventual data consistency without the need to put locks on data.
+
+The features provide a Coordinator, to which LRAs along with any corresponding method are registered, and the participant which allows annotated microservice methods to create a new or join an existing LRA.
+
+To start or join an LRA use the `@LRA` annotation. Depending on the type provided it may create a new LRA, join the existing LRA or continue outside one if none is provided. A full list of options can be found in the Microprofile Long Running Actions link:https://github.com/eclipse/microprofile-lra/blob/master/spec/src/main/asciidoc/microprofile-lra-spec.adoc[specifications].
+
+An example of always creating a new LRA when called would be:
+
+[source, java]
+----
+    @LRA(LRA.Type.REQUIRES_NEW)
+    @Path(“/start”)
+    @PUT
+    public void startAction(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId)
```

Comment:
```suggestion
    public Response startAction(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId)
```

This won't compile since it returns something for a void method. I suggest changing the return type to `Response` since that is what is returned in the method body.

Reply:
@yongja79 is this a correct change?

Reply:
Changed on Jason's OK 👍